oak_restricted_kernel/src/syscall/mod.rs

```diff
@@ -0,0 +1,155 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+mod channel;
+
+use alloc::boxed::Box;
+use core::{arch::asm, ffi::c_void};
+use oak_channel::Channel;
+use oak_restricted_kernel_runtime::{syscall::Errno, Syscall};
+use x86_64::{
+    registers::{
+        control::Efer,
+        model_specific::{EferFlags, LStar},
+    },
+    VirtAddr,
+};
+
+use self::channel::{syscall_fsync, syscall_read, syscall_write};
+
+pub fn enable_syscalls(channel: Box<dyn Channel>) {
+    channel::init(channel);
+
+    LStar::write(VirtAddr::new(syscall_entrypoint as *const fn() as u64));
+    unsafe {
+        Efer::update(|flags| flags.set(EferFlags::SYSTEM_CALL_EXTENSIONS, true));
+    }
+}
+
+// SysV ABI: arguments are in RDI, RSI, RDX, RCX; return value in RAX (which matches SYSRET!)
```

Comment:
super nit: I think this should either be converted to doc comments (with a bit of a description about the functions as well), or the comment should be inside the function. Similar for sycall_entrypoint.

Reply:
Done.